test/conftest.py

```diff
@@ -1,15 +1,20 @@
 # Copyright Contributors to the Pyro project.
 # SPDX-License-Identifier: Apache-2.0
 
-import pyro
-try:
-    from jax.config import config
+import numpy as np
 
-    config.update('jax_platform_name', 'cpu')
-except ModuleNotFoundError:
-    pass
+from funsor.util import get_backend
 
 
 def pytest_runtest_setup(item):
-    pyro.set_rng_seed(0)
-    pyro.enable_validation(True)
+    np.random.seed(0)
+    backend = get_backend()
+    if backend == "torch":
+        import pyro
+
+        pyro.set_rng_seed(0)
+        pyro.enable_validation(True)
+    elif backend == "jax":
+        from jax.config import config
+
+        config.update('jax_platform_name', 'cpu')
```

Comment:
I assume `set_backend()` should never be called by a test, rather backend should be set by `FUNSOR_BACKEND` environment variable. It might be worth documenting this or better adding a check like
```py
def _disallow_set_backend(*args):
    raise ValueError("set_backend() cannot be called during tests")

def pytest_runtest_setup(item):
    ...
    import funsor.util
    funsor.util.set_backend = _disallow_set_backend
```

Reply:
Thanks, I'll incorporate it in the next PR.